dropwizard-jersey/src/test/java/io/dropwizard/jersey/jackson/JsonProcessingExceptionMapperTest.java

```diff
@@ -97,7 +97,10 @@ public void returnsA400ForNonDeserializableRequestEntities() throws Exception {
 
     @Test
     public void returnsA400ForWrongInputType() throws Exception {
```

Comment:
I think I could see a few things to improve on Jackson side. `URI` in particular should probably internally signal `IllegalArgumentException` (or similar), and then bound to one of `JsonMappingException` subtypes, most likely `InvalidFormatException`. This could go in a patch since I think it's sort of a flaw at this point, underspecifying failure.

Similarly, "wrong token" style exceptions (`boolean` for Object) should probably be indicated differently, but they are little bit more difficult to figure out.

This is actually kind of work I was hoping to get started by changes that lead to these practical problems: reworking error handling. I have added some structure, but there is no uniformity yet. I think DW use case, as an example, could be a good driver in figuring out semantic approach to signalling failures.

Not sure how easy it is to divide things neatly into client/server buckets, however; it would seem that in many cases majority could be viewed as client problem, sending JSON that does not match expected structure.